src/SprykerEco/Zed/AvalaraTaxShipmentTax/Business/Calculator/AbstractShipmentAvalaraTaxCalculator.php

```diff
@@ -0,0 +1,162 @@
+use SprykerEco\Zed\AvalaraTaxShipmentTax\Business\Mapper\AvalaraLineItemMapper;
+use SprykerEco\Zed\AvalaraTaxShipmentTax\Dependency\Facade\AvalaraTaxShipmentTaxToMoneyFacadeInterface;
+use SprykerEco\Zed\AvalaraTaxShipmentTax\Dependency\Service\AvalaraTaxShipmentTaxToShipmentServiceInterface;
+use SprykerEco\Zed\AvalaraTaxShipmentTax\Dependency\Service\AvalaraTaxShipmentTaxToUtilEncodingServiceInterface;
+
+abstract class AbstractShipmentAvalaraTaxCalculator implements ShipmentAvalaraTaxCalculatorInterface
+{
+    /**
+     * @uses \Spryker\Shared\Shipment\ShipmentConfig::SHIPMENT_EXPENSE_TYPE
+     */
+    protected const SHIPMENT_EXPENSE_TYPE = 'SHIPMENT_EXPENSE_TYPE';
+
+    /**
+     * @var \SprykerEco\Zed\AvalaraTaxShipmentTax\Dependency\Facade\AvalaraTaxShipmentTaxToMoneyFacadeInterface
+     */
+    protected $moneyFacade;
+
+    /**
+     * @var \SprykerEco\Zed\AvalaraTaxShipmentTax\Dependency\Service\AvalaraTaxShipmentTaxToShipmentServiceInterface
+     */
+    protected $shipmentService;
+
+    /**
+     * @var \SprykerEco\Zed\AvalaraTaxShipmentTax\Dependency\Service\AvalaraTaxShipmentTaxToUtilEncodingServiceInterface
+     */
+    protected $utilEncodingService;
+
+    /**
+     * @param \SprykerEco\Zed\AvalaraTaxShipmentTax\Dependency\Facade\AvalaraTaxShipmentTaxToMoneyFacadeInterface $moneyFacade
+     * @param \SprykerEco\Zed\AvalaraTaxShipmentTax\Dependency\Service\AvalaraTaxShipmentTaxToShipmentServiceInterface $shipmentService
+     * @param \SprykerEco\Zed\AvalaraTaxShipmentTax\Dependency\Service\AvalaraTaxShipmentTaxToUtilEncodingServiceInterface $utilEncodingService
+     */
+    public function __construct(
+        AvalaraTaxShipmentTaxToMoneyFacadeInterface $moneyFacade,
+        AvalaraTaxShipmentTaxToShipmentServiceInterface $shipmentService,
+        AvalaraTaxShipmentTaxToUtilEncodingServiceInterface $utilEncodingService
+    ) {
+        $this->moneyFacade = $moneyFacade;
+        $this->shipmentService = $shipmentService;
+        $this->utilEncodingService = $utilEncodingService;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\CalculableObjectTransfer $calculableObjectTransfer
+     * @param \Generated\Shared\Transfer\AvalaraCreateTransactionResponseTransfer $avalaraCreateTransactionResponseTransfer
+     *
+     * @return \Generated\Shared\Transfer\CalculableObjectTransfer
+     */
+    abstract public function calculateTax(
+        CalculableObjectTransfer $calculableObjectTransfer,
+        AvalaraCreateTransactionResponseTransfer $avalaraCreateTransactionResponseTransfer
+    ): CalculableObjectTransfer;
+
+    /**
+     * @param \ArrayObject|\Generated\Shared\Transfer\AvalaraTransactionLineTransfer[] $avalaraTransactionLineTransfers
+     *
+     * @return \Generated\Shared\Transfer\AvalaraTransactionLineTransfer[][]
+     */
+    protected function getProductOptionAvalaraTransactionLineTransfersIndexedByItemSkuAndProductOptionSku(ArrayObject $avalaraTransactionLineTransfers): array
```

Comment:
* Fix the name.
* Change the name of the variables in the method.

Reply:
It is not even used in the module, just a copy-paste leftover, I'll remove it.

Reply:
Resolved.